File: c/binarySearch.c
#include <stdio.h>
#include <math.h>

// in binary search array must be sorted

int arr[] = {8, 13, 27, 38, 48, 54, 69, 78, 82, 100};
int item = 100;
int main()
{
    int left, right, middle;
    left = 0;
    right = (sizeof(arr) / sizeof(arr[0])) - 1;
    while (left <= right)
    {
        middle = floor((left + right) / 2);
        if (arr[middle] == item)
        {
            printf("Item found at index %d\n", middle);
            return 0;
        }
        else if (arr[middle] < item)
        {
            left = middle + 1;
        }
        else
        {
            right = middle - 1;
        }
    }
    printf("Item not found\n");
    return 0;
}

File: js/binarySearch.js
function binaryRecursive(arr, searchKey, left, right) {
  if (left > right) {
    return false;
  }
  let middle = Math.floor((left + right) / 2);
  if (arr[middle] === searchKey) {
    return true;
  }
  if (arr[middle] < searchKey) {
    return binaryRecursive(arr, searchKey, left, middle - 1);
  } else {
    return binaryRecursive(arr, searchKey, middle + 1, right);
  }
}
let arr = [1, 3, 5, 7, 8, 9];
let x = 6;

if (binaryRecursive(arr, x, 0, arr.length - 1)) {
  console.log("Element found");
} else {
  console.log("Element not found!");
}
